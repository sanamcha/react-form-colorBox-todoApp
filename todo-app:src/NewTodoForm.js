import React, { useState } from "react";
import {v1 as uuid} from "uuid";

const NewTodoForm = ({ createTodo }) => {
  const [list, setList] = useState("");

  const handleChange = e => {
    setList(e.target.value);
  };

  const input = e => {
    e.preventDefault();
    createTodo({ list, id: uuid() });
    setList("");
  };

  return (
    <div>
      <form onSubmit={ input }>
        <label htmlFor="list">TODO-LIST:</label>
        <input
          id="list"
          name="list"
          type="text"
          onChange={ handleChange }
          value={ list }
        />
        <button>Add Todo-List:</button>
      </form>
    </div>
  );
}

export default NewTodoForm;