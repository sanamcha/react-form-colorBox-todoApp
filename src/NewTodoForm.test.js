
import React from "react";
import { render, fireEvent } from "@testing-library/react";
import NewTodoForm from "./NewTodoForm";

it("renders without crashing", () => {
  render(<NewTodoForm />);
});

it("matches snapshot", () => {
  const { asFragment } = render(<NewTodoForm />);
  expect(asFragment()).toMatchSnapshot();
});


it("adds todo", () =>  {

  const { getByText } = render(<NewTodoForm createTodo={  jest.fn() } />);
  const createButton = getByText("Add Todo-List:");
  fireEvent.click(createButton);
 
});
