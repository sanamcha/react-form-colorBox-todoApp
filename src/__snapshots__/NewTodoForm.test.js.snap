// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<DocumentFragment>
  <div>
    <form>
      <label
        for="list"
      >
        TODO-LIST:
      </label>
      <input
        id="list"
        name="list"
        type="text"
        value=""
      />
      <button>
        Add Todo-List:
      </button>
    </form>
  </div>
</DocumentFragment>
`;
